pkg/elle/rw_register/rw_register.go

```diff
@@ -1 +1,902 @@
 package rwregister
+
+import (
+	"fmt"
+	"log"
+	"sort"
+
+	"github.com/pingcap/tipocket/pkg/elle/core"
+	"github.com/pingcap/tipocket/pkg/elle/txn"
+)
+
+// GraphOption ...
+type GraphOption struct {
+	LinearizableKeys bool //Uses realtime order
+	SequentialKeys   bool // Uses process order
+	WfrKeys          bool // Assumes writes follow reads in a txn
+}
+
+// key -> v -> Op
+type writeIdx map[string]map[int]core.Op
+
+// key -> {v1 -> [Op1, Op2, Op3...], v2 -> [Op4, Op5...]}
+type readIdx map[string]map[int][]core.Op
+
+// GCaseTp type aliases []core.Anomaly
+type GCaseTp []core.Anomaly
+
+// InternalConflict records a internal conflict
+type InternalConflict struct {
+	Op       core.Op
+	Mop      core.Mop
+	Expected core.Mop
+}
+
+// IAnomaly ...
+func (i InternalConflict) IAnomaly() {}
+
+// String ...
+func (i InternalConflict) String() string {
+	return fmt.Sprintf("(InternalConflict) Op: %s, mop: %s, expected: %s", i.Op, i.Mop.String(), i.Expected)
+}
+
+func internalOp(op core.Op) core.Anomaly {
+	dataMap := make(map[string]int)
+	for _, mop := range *op.Value {
+		if mop.IsWrite() {
+			for k, v := range mop.M {
+				vprt := v.(*int)
+				if vprt == nil {
+					panic("write value should not be nil")
+				}
+				dataMap[k] = *vprt
+			}
+		}
+		if mop.IsRead() {
+			for k, v := range mop.M {
+				vprt := v.(*int)
+				if vprt == nil {
```

Comment:
I think there is a logic in consistence, if vprt is nil, but `dataMap[k]` os not nil, it should report

Reply:
This should be report, but the nil read will fallback to `initMagicNumber`, so there should not be nil reads. Add a panic here.

Reply:
Where's the logic? how can I read `1919810`?

Reply:
ikuikuhaire?